test/v2/load/load_generator.go

```diff
@@ -257,9 +261,29 @@ func (l *LoadGenerator) disperseBlob(rand *random.TestRandom) (
 		return corev2.BlobKey{}, nil, nil, fmt.Errorf("failed to compute blob key: %w", err)
 	}
 
+	// Estimate gas for CheckDACert call
+	go l.estimateGasCheckDACertAsync(eigenDAV3Cert)
+
 	return blobKey, payload, eigenDACert, nil
 }
 
+// estimateGasCheckDACertAsync performs gas estimation in a separate goroutine to avoid blocking blob dispersal.
+func (l *LoadGenerator) estimateGasCheckDACertAsync(eigenDAV3Cert *coretypes.EigenDACertV3) {
+	l.gasEstimationLimiter <- struct{}{}
+	defer func() {
+		<-l.gasEstimationLimiter
+	}()
+
+	gasTimeout := time.Duration(l.config.GasEstimationTimeout) * time.Second
+	ctx, cancel := context.WithTimeout(context.Background(), gasTimeout)
```

Comment:
```suggestion
	ctx, cancel := context.WithTimeout(l.ctx, gasTimeout)
```

Reply:
Generally prefer not to store contexts in structs, following arguments in https://go.dev/blog/context-and-structs, but in any case the ctx does exist so we should def use it instead of context.Background

@cody-littley did you have a good reason for storing this one? Is it because its the main context that is used to cancel all goroutines when a sigterm signal is received?

Reply:
suggested change included in [85e3469](https://github.com/Layr-Labs/eigenda/pull/1890/commits/85e3469d534a655ed598270c9e9f43fcbcafae39)